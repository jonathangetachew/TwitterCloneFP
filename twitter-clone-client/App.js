import { AppLoading } from "expo";
import { Asset } from "expo-asset";
import * as Font from "expo-font";
import React, { useState } from "react";
import {
  Platform,
  StatusBar,
  SafeAreaView,
  StyleSheet,
  View
} from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { ThemeProvider } from "react-native-elements";
// Added this imports
import { Provider } from "react-redux";
import { persistStore } from "redux-persist";
import { PersistGate } from "redux-persist/integration/react";
import storage from "./src/redux/store";
import rootSaga from "./src/redux/sagas";
import AppNavigator from "./src/navigation/AppNavigator";

const persistor = persistStore(storage.getStore());

storage.getSagaMiddleware().run(rootSaga);

export default function App(props) {
  const [isLoadingComplete, setLoadingComplete] = useState(false);
  let Content;

  if (!isLoadingComplete && !props.skipLoadingScreen) {
    Content = (
      <AppLoading
        startAsync={loadResourcesAsync}
        onError={handleLoadingError}
        onFinish={() => handleFinishLoading(setLoadingComplete)}
      />
    );
  } else {
    Content = (
      <View style={styles.container}>
        {Platform.OS === "ios" && <StatusBar barStyle="default" />}
        <AppNavigator />
      </View>
    );
  }

  return (
    <Provider store={storage.getStore()}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider>
          <SafeAreaView style={{ flex: 1, backgroundColor: "#fff" }}>
            {Content}
          </SafeAreaView>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  );
}

async function loadResourcesAsync() {
  await Promise.all([
    Asset.loadAsync([require("./assets/images/logo.png")]),
    Font.loadAsync({
      // This is the font that we are using for our tab bar
      ...Ionicons.font,
      // We include SpaceMono because we use it in HomeScreen.js. Feel free to
      // remove this if you are not using it in your app
      "space-mono": require("./assets/fonts/SpaceMono-Regular.ttf")
    })
  ]);
}

function handleLoadingError(error) {
  // In this case, you might want to report the error to your error reporting
  // service, for example Sentry
  console.warn(error);
}

function handleFinishLoading(setLoadingComplete) {
  setLoadingComplete(true);
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  }
});
