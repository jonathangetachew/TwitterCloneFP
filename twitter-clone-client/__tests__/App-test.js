import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

jest.mock('react-redux', () => ({
  Provider: 'Provider',
}));

jest.mock('redux-persist/integration/react', () => ({
  PersistGate: 'PersistGate',
}));

jest.mock('react-native-elements', () => ({
  ThemeProvider: 'ThemeProvider'
}));

jest.mock('../src/navigation/AppNavigator', () => 'AppNavigator');

describe('App', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`renders the loading screen`, () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`renders the root without loading screen`, () => {
    const tree = renderer.create(<App skipLoadingScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
