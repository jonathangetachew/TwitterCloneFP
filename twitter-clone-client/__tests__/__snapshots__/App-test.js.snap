// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ThemeProvider>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
        }
      >
        <AppLoading
          onError={[Function]}
          onFinish={[Function]}
          startAsync={[Function]}
        />
      </RCTSafeAreaView>
    </ThemeProvider>
  </PersistGate>
</Provider>
`;

exports[`App renders the root without loading screen 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ThemeProvider>
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <AppNavigator />
        </View>
      </RCTSafeAreaView>
    </ThemeProvider>
  </PersistGate>
</Provider>
`;
